src/dodrio_berry_picking.c

```diff
@@ -1,870 +1,2437 @@
+            case 5:
+                gUnknown_203F3E0->unk4A[0][3]++;
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            case 6:
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            case 7:
+                gUnknown_203F3E0->unk4A[1][3]++;
+                gUnknown_203F3E0->unk4A[2][3]++;
+                break;
+            case 8:
+                gUnknown_203F3E0->unk4A[2][3]++;
+                break;
+            }
+            break;
+        case 3:
+            switch (arg1)
+            {
+            case 2:
+                gUnknown_203F3E0->unk4A[1][3]++;
+                gUnknown_203F3E0->unk4A[2][3]++;
+                break;
+            case 3:
+                gUnknown_203F3E0->unk4A[2][3]++;
+                break;
+            case 4:
+                gUnknown_203F3E0->unk4A[2][3]++;
+                gUnknown_203F3E0->unk4A[0][3]++;
+                break;
+            case 5:
+                gUnknown_203F3E0->unk4A[0][3]++;
+                break;
+            case 6:
+                gUnknown_203F3E0->unk4A[0][3]++;
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            case 7:
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            }
+            break;
+        case 2:
+            switch (arg1)
+            {
+            case 3:
+                gUnknown_203F3E0->unk4A[0][3]++;
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            case 4:
+                gUnknown_203F3E0->unk4A[0][3]++;
+                break;
+            case 5:
+                gUnknown_203F3E0->unk4A[0][3]++;
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            case 6:
+                gUnknown_203F3E0->unk4A[1][3]++;
+                break;
+            }
+            break;
+        }
+        break;
+    }
+}
 
-/*static*/ const struct WindowTemplate sUnknown_8475674[] =
+static void sub_8152F94(bool32 arg0)
 {
+    if (gUnknown_203F3E0->unk24 != 5)
+        return;
+
+    if (arg0 == TRUE)
     {
-        .bg = 0,
-        .tilemapLeft = 1,
-        .tilemapTop = 1,
-        .width = 28,
-        .height = 3,
-        .paletteNum = 13,
-        .baseBlock = 0x13,
-    },
+        if (++gUnknown_203F3E0->unk112 > gUnknown_203F3E0->unk114)
+            gUnknown_203F3E0->unk114 = gUnknown_203F3E0->unk112;
+        if (gUnknown_203F3E0->unk112 > 9999)
+            gUnknown_203F3E0->unk112 = 9999;
+    }
+    else
     {
-        .bg = 0,
-        .tilemapLeft = 1,
-        .tilemapTop = 6,
-        .width = 28,
-        .height = 13,
-        .paletteNum = 13,
-        .baseBlock = 0x67,
+        if (gUnknown_203F3E0->unk112 > gUnknown_203F3E0->unk114)
+            gUnknown_203F3E0->unk114 = gUnknown_203F3E0->unk112;
+        gUnknown_203F3E0->unk112 = 0;
     }
-};
+}
 
-/*static*/ const struct WindowTemplate sUnknown_8475684 =
+static void 
```

Comment:
It's different, because here they're compared as unsigned values.

Reply:
github mess up line numbers